red4ext-sys/src/interop.rs

```diff
@@ -84,7 +84,104 @@ unsafe impl ExternType for TweakDBID {
     type Kind = cxx::kind::Trivial;
 }
 
-#[derive(Debug, Clone)]
+/// see [its C++ representation](https://github.com/WopsS/RED4ext.SDK/blob/master/include/RED4ext/NativeTypes.hpp#L105)
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+#[repr(C)]
+pub struct ItemID {
+    id: TweakDBID,
+    seed: Seed,
+    counter: u16,
+    /// also called [`unknown` in CET](https://wiki.redmodding.org/cyber-engine-tweaks/functions/special-types#toitemid)
+    structure: gamedataItemStructure,
+    /// also called [`maybe_type` in CET](https://wiki.redmodding.org/cyber-engine-tweaks/functions/special-types#toitemid)
+    flags: gameEItemIDFlag,
+}
+
+impl ItemID {
+    pub fn new() -> Self {
+        Self {
+            ..Default::default()
+        }
+    }
+    pub fn new_from_tdbid(id: &TweakDBID, seed: impl Into<Seed>) -> Self {
```

Comment:
This should accept `TweakDBID` directly without a reference (constructors shouldn't force a clone if it's not necessary)